src/span.js

```diff
@@ -192,19 +191,19 @@ export default class Span {
      * Adds a log event, or payload to a span.
      *
      * @param {Object} fields - an object that represents the fields to log.
-     * @param {number} [fields.timestamp] - the starting timestamp of a span.
+     * @param {number} [timestamp] - the starting timestamp of a span.
      * @param {string} [fields.event] - a string message to be logged on a span.
      * @param {string} [fields.payload] - an object to be logged on a span as a json string.
      **/
-    log(fields: any): void {
+    log(fields: any, timestamp: ?number): void {
         if (this._spanContext.isSampled()) {
             if (!fields.event && !fields.payload) {
                 this._logger.error('log must be passed either an event of type string, or a payload of type object');
                 return;
             }
 
             if (!fields.timestamp) {
-                fields.timestamp = Utils.getTimestampMicros();
+                fields.timestamp = timestamp || Utils.getTimestampMicros();
             }
 
             let value = fields.event;
```

Comment:
3 lines below - you simply want to push `fields` hash, not a new hash. There's nothing special about event/payload keys anymore.